### DISCUSSION TITLE
Errors when using args in python script

### DISCUSSION THREAD

**Discussion by lukik** [dashingsoft/pyarmor#1865] [Q&A] [answered]
I am trying to use the [args feature mentioned in the documentation](https://pyarmor.readthedocs.io/en/latest/tutorial/installation.html#run-pyarmor-from-python-script) but it keeps failing with different errors depending on the parameters passed. 

As an example, the following command targeting a specific platform is successful on terminal or when using python's subprocess but if fails when using `args`

**Terminal - Works Ok:** 
`$pyarmor gen --platform alpine.x86_64 myscript.py`

**Python Script using subprocess - Works Ok:** 
```
import subprocess
file_to_obfuscate = f"{project_dir}/myscript.py"
full_command = f"pyarmor gen --platform alpine.x86_64 {file_to_obfuscate}"
subprocess.run(
    full_command,
    shell=True,
    cwd=project_directory,
    executable="/bin/bash"
)
```

**Python script using args - Fails:**

```
file_to_obfuscate = f"{project_dir}/myscript.py"
args = [
    'gen',
    '--platform alpine.x86_64',
    file_to_obfuscate
]
main_entry(args)
```

Error: `pyarmor.cli.CliError: no found input "--platform alpine.x86_64`

What could I be doing wrong? 

**Pyarmor Details:**
Version: latest (details below)
License: Trial
- python = "^3.11"
- pyarmor = "^8.5.10"
- pyarmor-cli = "^8.5.10"
- pyarmor-cli-core-windows = "^6.5.3"
- pyarmor-cli-core-linux = "^6.5.3"
- pyarmor-cli-core-alpine = "6.5.3"
- pyarmor-cli-core = "^6.5.3"
- pyarmor-cli-runtime = "^3.2.5"

**Comment by jondy** [comment, marked as answer]
Try this: `args = ['gen', '--platform', 'alpine.x86_64', file_to_obfuscate]`

**Comment by lukik** [comment]
Yep. That worked. Thanks!

It now looks so obvious now that you post it. Not sure if it could help others if such an example was added to the docs.

**Comment by jondy** [comment]
Good idea. Check this patch a6a006edf1ac10ae8dcdf9084848de57823d0cad

**Comment by lukik** [comment]
That will for sure help future users get it right first time. Thanks!
